From 39d6b91a2b1ade263d59d5b749921dcf736c3946 Mon Sep 17 00:00:00 2001
Message-Id: <39d6b91a2b1ade263d59d5b749921dcf736c3946.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Mon, 7 Jul 2014 14:16:48 +0100
Subject: [PATCH 138/312] FOR_UPSTREAM [VPG]: drm/i915: Add
 WaHdcDisableFetchWhenMasked:bdw

Set desired default value for HDCCHICKEN register for BDW platform.

Issue: VIZ-4086
Change-Id: Id39776937a10ef339cea54662004110e414e3ab7
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    1 +
 drivers/gpu/drm/i915/intel_pm.c |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 4671ce2..97f70a1 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4737,6 +4737,7 @@ enum punit_power_well {
 /* GEN8 chicken */
 #define HDC_CHICKEN0				0x7300
 #define  HDC_FENCE_DESTINATION_TO_SLM_DISABLE	(1<<14)
+#define  HDC_DONOT_FETCH_MEM_WHEN_MASKED		(1<<11)
 #define  HDC_FORCE_NON_COHERENT			(1<<4)
 
 /* WaProgramL3SqcReg1Default */
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 3aace0c..6dc3b37 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5572,10 +5572,12 @@ static void gen8_init_clock_gating(struct drm_device *dev)
 	 * workaround for for a possible hang in the unlikely event a TLB
 	 * invalidation occurs during a PSD flush.
 	 * WaDisableFenceDestinationToSLM:bdw
+	 * WaHdcDisableFetchWhenMasked:bdw
 	 */
 	I915_WRITE(HDC_CHICKEN0,
 		   I915_READ(HDC_CHICKEN0) |
 		   _MASKED_BIT_ENABLE(HDC_FENCE_DESTINATION_TO_SLM_DISABLE) |
+		   _MASKED_BIT_ENABLE(HDC_DONOT_FETCH_MEM_WHEN_MASKED) |
 		   _MASKED_BIT_ENABLE(HDC_FORCE_NON_COHERENT));
 
 	/* WaVSRefCountFullforceMissDisable:bdw */
-- 
1.7.9.5

